src/Microsoft.VisualStudio.SlnGen/Tasks/SlnGenToolTask.cs

```diff
@@ -139,6 +145,22 @@ protected override string GenerateCommandLineCommands()
             commandLineBuilder.AppendSwitchIfNotNull("--solutionfile:", GetPropertyValue(MSBuildPropertyNames.SlnGenSolutionFileFullPath));
             commandLineBuilder.AppendSwitchIfNotNull("--property:", globalProperties.Count == 0 ? null : string.Join(";", globalProperties.Select(i => $"{i.Key}={i.Value}")));
 
+            // The vsversion switch can be passed without a value to ask slngen to deduce the desired version.
+            // We cannot differentiate between a property that is set to empty and a property that doesn't
+            // exist, so we use the special value "default" for SlnGenVSVersion to mark that the
+            // flag should be given without a value.
+            string vsVersion = GetPropertyValue(MSBuildPropertyNames.SlnGenVSVersion);
+            if (vsVersion == SlnGenVSVersionDefault)
```

Comment:
Let's do a case-insensitive comparison here please, just in case someone uses a value like `Default`.

```suggestion
            if (string.Equals(vsVersion, SlnGenVSVersionDefault, StringComparison.OrdinalIgnoreCase)
```